lib/client/bst-virtual-alexa.ts

```diff
@@ -0,0 +1,204 @@
+import {Global} from "../core/global";
+import {SkillContext, VirtualAlexa} from "virtual-alexa";
+import * as fs from "fs";
+
+/**
+ * Programmatic interface for interacting with the Virtual Alexa.
+ *
+ */
+export class BSTVirtualAlexa {
+    public static DefaultIntentSchemaLocation = "speechAssets/IntentSchema.json";
+    public static DefaultSampleUtterancesLocation = "speechAssets/SampleUtterances.txt";
+    public static DefaultInteractionModelLocation = "speechAssets/InteractionModel.json";
+
+    private virtualAlexa: VirtualAlexa = null;
+    private interactionModelProvided: boolean = false;
+    private sampleUtterancesProvided: boolean = false;
+    private intentSchemaProvided: boolean = false;
+
+    private static FileTypes = {
+        InterationModel: "Interaction Model",
+        IntentSchema: "Intent Schema",
+        SampleUtterances: "Sample Utterances"
+    };
+
+    /**
+     * Creates a new Alexa emulator
+     * @param skillURL The URL the skill is listening that this emulator should interact with
+     * @param interactionModel The path to the interaction model file - defaults to {@link BSTAlexa.DefaultInteractionModelLocation}.
+     * @param intentSchemaFile The path to the intent schema file - defaults to {@link BSTAlexa.DefaultIntentSchemaLocation}.
+     * @param sampleUtterancesFile The path to the samples utterances file - defaults to {@link BSTAlexa.DefaultSampleUtterancesLocation}.
+     * @param applicationID The application ID. Just makes one up if none is defined.
+     */
+    public constructor(private skillURL: string,
+                       private interactionModel?: string,
+                       private intentSchemaFile?: string,
+                       private sampleUtterancesFile?: string,
+                       private applicationID?: string) {
+        if ((intentSchemaFile || sampleUtterancesFile) && interactionModel) {
+            console.error("Interaction Model and Intent Schema Files shouldn't be sent together. Chose one of those");
```

Comment:
"The Interaction Model and Intent Schema Files should not both be specified. It should be one or the other."